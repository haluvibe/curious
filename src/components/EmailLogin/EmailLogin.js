// @flow
import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Button from "material-ui/Button";
import TextField from "material-ui/TextField";
import { withStyles } from "material-ui/styles";

const styles = theme => ({
  container: {
    display: "flex",
    flexWrap: "wrap"
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 200
  },
  button: {
    margin: theme.spacing.unit
  }
});

const EmailLogin = ({
  t,
  classes,
  onEmailChange,
  onPasswordChange,
  emailValue,
  passwordValue,
  onSubmit,
  emailError
}) => {
  return (
    <form
      className={classes.container}
      noValidate
      autoComplete="off"
      onSubmit={event => onSubmit(event)}
    >
      <div className="form__row email-login-form__row">
        <TextField
          id="email"
          label="email address"
          helperText={emailError}
          type="email"
          margin="normal"
          className={classes.textField}
          inputRef={c => this.email = c}
          error={!!emailError}
          value={emailValue}
          onChange={() => onEmailChange(this.email.value)}
        />
        <br />
        <TextField
          id="password"
          label="password"
          inputRef={c => this.password = c}
          margin="normal"
          value={passwordValue}
          type="password"
          autoComplete="current-password"
          className={classes.textField}
          onChange={() => onPasswordChange(this.password.value)}
        />
      </div>
      <div className="form__row email-login-form__row">
        <Button label="Primary" variant="raised" className={classes.button}>
          <input type="submit" value="Login" />
        </Button>
      </div>
      <div className="form__row email-login-form__row">
        <Button label="Secondary" variant="raised" className={classes.button}>
          <Link to={"/forgot-password"}>Forgot password?</Link>
        </Button>
      </div>
    </form>
  );
};

export default withStyles(styles)(EmailLogin);

EmailLogin.propTypes = {
  onEmailChange: PropTypes.func.isRequired,
  onPasswordChange: PropTypes.func.isRequired,
  emailValue: PropTypes.string,
  passwordValue: PropTypes.string,
  onSubmit: PropTypes.func.isRequired
};

EmailLogin.defaultProps = {
  emailValue: "",
  passwordValue: "",
  t: () => {},
  onEmailChange: () => {},
  onPasswordChange: () => {},
  emailValue: "",
  passwordValue: "",
  onSubmit: () => {}
};
