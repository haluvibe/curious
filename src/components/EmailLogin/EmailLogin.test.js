import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import EmailLogin from "./EmailLogin";

test("EmailLogin component should render as expected", () => {
    const component = shallow(<EmailLogin />),
        tree = toJson(component.dive());
    expect(tree).toMatchSnapshot();
});
