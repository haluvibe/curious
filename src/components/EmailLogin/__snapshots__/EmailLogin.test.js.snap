// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailLogin component should render as expected 1`] = `
<form
  autoComplete="off"
  className="EmailLogin-container-1"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="form__row email-login-form__row"
  >
    <TextField
      className="EmailLogin-textField-2"
      error={false}
      id="email"
      inputRef={[Function]}
      label="email address"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      type="email"
      value=""
    />
    <br />
    <TextField
      autoComplete="current-password"
      className="EmailLogin-textField-2"
      id="password"
      inputRef={[Function]}
      label="password"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      type="password"
      value=""
    />
  </div>
  <div
    className="form__row email-login-form__row"
  >
    <WithStyles(Button)
      className="EmailLogin-button-3"
      label="Primary"
      variant="raised"
    >
      <input
        type="submit"
        value="Login"
      />
    </WithStyles(Button)>
  </div>
  <div
    className="form__row email-login-form__row"
  >
    <WithStyles(Button)
      className="EmailLogin-button-3"
      label="Secondary"
      variant="raised"
    >
      <Link
        replace={false}
        to="/forgot-password"
      >
        Forgot password?
      </Link>
    </WithStyles(Button)>
  </div>
</form>
`;
